### PR TITLE
add libpq for multi-arch build configuration

### PR DESCRIPTION
**Pull request by arowser**
add libpq for multi-arch build configuration

Fix https://github.com/ElementsProject/lightning/issues/7649

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -94,7 +94,8 @@ RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.9 1
 RUN pip3 install --upgrade pip setuptools wheel
 
 RUN wget -q https://zlib.net/fossils/zlib-1.2.13.tar.gz -O zlib.tar.gz && \
-    wget -q https://www.sqlite.org/2019/sqlite-src-3290000.zip -O sqlite.zip
+    wget -q https://www.sqlite.org/2019/sqlite-src-3290000.zip -O sqlite.zip \
```
Comment (s373nZ) [inline Dockerfile:97]: When I tried to build on x86, just to make a sanity check, this line fails for me. Looks like it needs an `&&` to concat the commands?

**Review by s373nZ — CHANGES_REQUESTED** [review]
My prior comment was marked as resolved, but there were no code changes or explanation? Still seeing a build error. The rest of the PR looks like it should work, though.

Can you explain how you're testing this?

**Comment by arowser** [comment]
What's your build error?
The dockerfile use to crossbuild with qemu, but the libq not exists on arm
environment, so I added the build steps for the  libq library.

**Comment by s373nZ** [comment]
> What's your build error? The dockerfile use to crossbuild with qemu, but the libq not exists on arm environment, so I added the build steps for the libq library.

The error is related to `wget` downloading the source code, which produces the `docker build` error:

```
 ERROR [base-builder  6/12] RUN wget -q https://zlib.net/fossils/zlib-1.2.13.tar.gz -O zlib.tar.  8.9s
------
 > [base-builder  6/12] RUN wget -q https://zlib.net/fossils/zlib-1.2.13.tar.gz -O zlib.tar.gz &&     wget -q https://www.sqlite.org/2019/sqlite-src-3290000.zip -O sqlite.zip     wget -q https://ftp.postgresql.org/pub/source/v17.1/postgresql-17.1.tar.gz -O postgres.tar.gz:
------
Dockerfile:96
--------------------
  95 |
  96 | >>> RUN wget -q https://zlib.net/fossils/zlib-1.2.13.tar.gz -O zlib.tar.gz && \
  97 | >>>     wget -q https://www.sqlite.org/2019/sqlite-src-3290000.zip -O sqlite.zip \
  98 | >>>     wget -q https://ftp.postgresql.org/pub/source/v17.1/postgresql-17.1.tar.gz -O postgres.tar.gz
  99 |
--------------------
ERROR: failed to solve: process "/bin/sh -c wget -q https://zlib.net/fossils/zlib-1.2.13.tar.gz -O zlib.tar.gz &&     wget -q https://www.sqlite.org/2019/sqlite-src-3290000.zip -O sqlite.zip     wget -q https://ftp.postgresql.org/pub/source/v17.1/postgresql-17.1.tar.gz -O postgres.tar.gz" did not complete successfully: exit code: 4
```

As I mentioned in  my initial review comment, it should be an easy fix, adding the `&&` to the end of that line. Curious how you've tested this, though, if that is breaking.

**Comment by s373nZ** [comment]
Additionally, this PR needs the commit message updated to contain a `Changelog-*:` notation in the message body as described in [this documentation](https://docs.corelightning.org/docs/coding-style-guidelines#changelog-entries-in-commit-messages) to avoid the build failure here:

https://github.com/ElementsProject/lightning/actions/runs/11883516546/job/33110390750?pr=7827

**Comment by arowser** [comment]
The errors have been resolved as you mention, if you have any other build
issues please let me know.

**Comment by s373nZ** [comment]
Not seeing any new changes. Did you forget to `git push` your updated code?

**Pull request closed without merging**